rinja/src/filters/alloc.rs

```diff
@@ -0,0 +1,657 @@
+use alloc::string::String;
+use core::fmt::{self, Write};
+
+use super::MAX_LEN;
+use super::escape::HtmlSafeOutput;
+use crate::Result;
+
+/// Return an ephemeral `&str` for `$src: impl fmt::Display`
+///
+/// If `$str` is `&str` or `String`, this macro simply passes on its content.
+/// If it is neither, then the formatted data is collection into `&buffer`.
+///
+/// `return`s with an error if the formatting failed.
+macro_rules! try_to_str {
+    ($src:expr => $buffer:ident) => {
+        match format_args!("{}", $src) {
+            args => {
+                if let Some(s) = args.as_str() {
+                    s
+                } else {
+                    $buffer = String::new();
+                    $buffer.write_fmt(args)?;
+                    &$buffer
+                }
+            }
+        }
+    };
+}
+
+/// Formats arguments according to the specified format
+///
+/// The *second* argument to this filter must be a string literal (as in normal
+/// Rust). The two arguments are passed through to the `format!()`
+/// [macro](https://doc.rust-lang.org/stable/std/macro.format.html) by
+/// the Rinja code generator, but the order is swapped to support filter
+/// composition.
+///
+/// ```ignore
+/// {{ value|fmt("{:?}") }}
+/// ```
+///
+/// ```
+/// # #[cfg(feature = "code-in-doc")] {
+/// # use rinja::Template;
+/// /// ```jinja
+/// /// <div>{{ value|fmt("{:?}") }}</div>
+/// /// ```
+/// #[derive(Template)]
+/// #[template(ext = "html", in_doc = true)]
+/// struct Example {
+///     value: (usize, usize),
+/// }
+///
+/// assert_eq!(
+///     Example { value: (3, 4) }.to_string(),
+///     "<div>(3, 4)</div>"
+/// );
+/// # }
+/// ```
+///
+/// Compare with [format](./fn.format.html).
+pub fn fmt() {}
+
+/// Formats arguments according to the specified format
+///
+/// The first argument to this filter must be a string literal (as in normal
+/// Rust). All arguments are passed through to the `format!()`
+/// [macro](https://doc.rust-lang.org/stable/std/macro.format.html) by
+/// the Rinja code generator.
+///
+/// ```ignore
+/// {{ "{:?}{:?}"|format(value, other_value) }}
+/// ```
+///
+/// ```
+/// # #[cfg(feature = "code-in-doc")] {
+/// # use rinja::Template;
+/// /// ```jinja
+/// /// <div>{{ "{:?}"|format(value) }}</div>
+/// /// ```
+/// #[derive(Template)]
+/// #[template(ext = "html", in_doc = true)]
+/// struct Example {
+///     value: (usize, usize),
+/// }
+///
+/// assert_eq!(
+///     Example { value: (3, 4) }.to_string(),
+///     "<div>(3, 4)</div>"
+/// );
+/// # }
+/// ```
+///
+/// Compare with [fmt](./fn.fmt.html).
+pub fn format() {}
+
+/// Replaces line breaks in plain text with appropriate HTML
```

Comment:
Missing a dot at the end of this sentence.

Reply:
I kept the comments as they are in the current master, I just split the file in two. I think it's easier to follow the individual commits than looking at the whole diff.

Reply:
I'll make a typo pass later on then. ;)